src/jsx.d.ts

```diff
@@ -1630,34 +1642,47 @@ export namespace JSXInternal {
 		| 'none'
 		| 'note'
 		| 'option'
+		| 'paragraph'
 		| 'presentation'
 		| 'progressbar'
 		| 'radio'
 		| 'radiogroup'
+		| 'range'
 		| 'region'
+		| 'roletype'
 		| 'row'
 		| 'rowgroup'
 		| 'rowheader'
 		| 'scrollbar'
 		| 'search'
 		| 'searchbox'
+		| 'section'
+		| 'sectionhead'
+		| 'select'
 		| 'separator'
 		| 'slider'
 		| 'spinbutton'
 		| 'status'
+		| 'strong'
+		| 'structure'
+		| 'subscript'
+		| 'superscript'
 		| 'switch'
 		| 'tab'
 		| 'table'
 		| 'tablist'
 		| 'tabpanel'
 		| 'term'
 		| 'textbox'
+		| 'time'
 		| 'timer'
 		| 'toolbar'
 		| 'tooltip'
 		| 'tree'
 		| 'treegrid'
 		| 'treeitem'
+		| 'widget'
+		| 'window'
 		| 'none presentation';
```

Comment:
This is a bit of a weird one, do we need to support it?

[`none` is meant to be an alias of `presentation`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/none_role), so I don't understand why someone would set both intentionally.

Reply:
Looks intentional: https://github.com/preactjs/preact/pull/3910/commits/8223d002e3784c2a2bcfe725d8ccd2ce17aebafb

Good enough for me!

Reply:
Turns out it's a fallback: https://www.w3.org/TR/wai-aria-1.1/#presentation-0

> Note regarding the ARIA 1.1 none role.[§](https://www.w3.org/TR/wai-aria-1.1/#role-presentation-note-none)

> In ARIA 1.1, the working group introduced none as a synonym to the presentation role, due to author confusion surrounding the intended meaning of the word "presentation" or "presentational." Many individuals erroneously consider role="presentation" to be synonymous with aria-hidden="true", and we believe role="none" conveys the actual meaning more unambiguously.

> **Until implementations include sufficient support for role="none", web authors are advised to use the presentation role alone role="presentation" or redundantly as a fallback to the none role role="none presentation".**